### ISSUE TITLE
how to save a 4.3 version pmml model using jpmml-xgboost code?

### ISSUE THREAD

**Issue by andyhascutename** [jpmml/jpmml-xgboost#69]
the current jpmml-xgboost code only output pmml version4.4, how can I save an older version(4.3) using jpmml-xgboost code?

**Comment by vruusmann** [comment]
> the current jpmml-xgboost code only output pmml version 4.4

This is a feature, not a bug.

PMML 4.4 makes it possible to generate smaller and more elegant markup.

> how can I save an older version(4.3) using jpmml-xgboost code?

The JPMML-XGBoost library currently supports only one output representation, which is "latest and most efficient". It is possible to use standalone tools to "expand" this representation into legacy and/or less efficient representations. This version inter-conversion is a job for a specialized tool. There is no point in integrating such functionality into the main JPMML-XGBoost codebase.

TLDR: These are your options for getting XGBoost models in PMML schema version 4.3:

- Use some JPMML-XGBoost legacy version (say, some three-four year old version), which was still generating the 4.3 representation (because the 4.4 representation didn't exist yet).
- Perform the conversion from 4.4 to 4.3 **manually**. First and foremost, change the PMML namespace URI, and try to feed the PMML document into your target software. Maybe it gets accepted as is. If not, see what the next complaint is, fix it, and repeat the process.
- Perform the same conversion automatically. You can build such a tool yourself (after gaining some proficiency by performing the above activities for a number of times), or you can purchase such a tool from my company.

**Comment by vruusmann** [comment]
The `sklearn2pmml.sklearn2pmml` utility method provides the `pmml_schema` parameter starting from SkLearn2PMML 0.110.0.

This utility function accepts XGBoost Booster objects, and its Scikit-Learn estimator wrapper objects.
